src/v/datalake/coordinator/coordinator.cc

```diff
@@ -47,11 +54,151 @@ std::ostream& operator<<(std::ostream& o, coordinator::errc e) {
     }
 }
 
+void coordinator::start() {
+    ssx::spawn_with_gate(gate_, [this] { return run_until_abort(); });
+}
+
 ss::future<> coordinator::stop_and_wait() {
     as_.request_abort();
+    leader_cond_.broken();
+    if (term_as_.has_value()) {
+        term_as_->get().request_abort();
+    }
     return gate_.close();
 }
 
+ss::future<> coordinator::run_until_abort() {
+    while (!as_.abort_requested()) {
+        auto& raft = *stm_->raft();
+        if (!raft.is_leader()) {
+            bool shutdown = false;
+            auto leader = co_await ss::coroutine::as_future(
+              leader_cond_.wait());
```

Comment:
>next the `wait` is invoked on condition variable but it was already signaled

Wouldn't this `wait()` in your example be avoided due to check for leadership on L73?

Reply:
cv waits will remember signals that were seen when not waiting. https://github.com/redpanda-data/seastar/blob/4a3406ce4071f0bc472233c4c669582ca527db89/tests/unit/condition_variable_test.cc#L44-L50

>next the wait is invoked on condition variable but it was already signaled

This would result in the loop proceeding.